sycl/tools/CMakeLists.txt

```diff
@@ -8,6 +8,12 @@ add_dependencies(get_device_count_by_type ocl-headers ocl-icd)
 add_definitions(-D MIN_INTEL_OCL_GPU_VERSION=\\"18.47.11882\\")
 add_definitions(-D MIN_INTEL_OCL_CPU_VERSION=\\"18.1.0.0901\\",\\"7.6.0.1202\\")
 
+if(MSVC)
+# __SYCL_BUILD_SYCL_DLL is used only to build SYCL library.
+# All users of SYCL library must not use this macro.
+remove_definitions(-D__SYCL_BUILD_SYCL_DLL)
```

Comment:
Much better approach is to move the logic building SYCL runtime library to `source/CMakeLists.txt` and set this definition for library only.

Reply:
Good idea, but currently the file source/CMakeLists.txt does not exist. 
How much of the logic do you suggest moving to a newly created file? Only add_definition(-D__SYCL_BUILD_SYCL_DLL)?

Reply:
Hmm..., I would rather keep the existing sycl/CMakeLists.txt for now. The intention of this patch was to fix a problem in PI, not restructuring the build/make files.

Reply:
> The intention of this patch was to fix a problem in PI, not restructuring the build/make files.

Because build/make files structure is wrong, you have to implement "bad" fix for "a problem in PI".

I prefer implementing good solution from start and I think it's right time to do the refactoring of build/make files first and only after that fix "a problem in PI".
We need to move the logic building SYCL runtime library (i.e. libsycl) to sycl/CMakeLists.txt. This should be an easy task IMHO.

Reply:
I re-based this patch. Thank you for adding sycl/source/CMakeLists.txt